### PR TITLE
Implemented parity_set

### PR DESCRIPTION
**Pull request by jordipainan**
- `parity_set` tested on a dev node
- everything working except `parity_setTransactionsLimit` (issue opened)
- `setTxGasLimit` is deprecated (node ignores the request, but stills in wiki so we can remove when docs upgrade)
- `setMinGasPrice` is deprecated (node ignores the request, but stills in wiki so we can remove when docs upgrade)

All methods are implemented but 1 not working and 2 deprecated. I took the approach to implement everything but maybe you think its better to remove all 3. We can remove code so its fine :)

A big thank you to @CPerezz helping me with return values ^^

### CODE REVIEW

File: src/api/mod.rs [hunk 1]
```diff
@@ -18,6 +19,7 @@ pub use self::parity_accounts::ParityAccounts;
 pub use self::personal::Personal;
 pub use self::web3::Web3 as Web3Api;
 pub use self::traces::Traces;
+pub use self::parity_set::ParitySet;
```
Comment (tomusdrw) [inline src/api/mod.rs]: Please keep sorted.

File: src/api/parity_set.rs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+#[derive(Debug, Clone)]
+/// `Parity_Set` Specific API
+pub struct ParitySet<T> {
+    transport: T
+}
+
+impl<T: Transport> Namespace<T> for ParitySet<T> {
+    fn new(transport: T) -> Self
+        where
+        Self: Sized,
```
Comment (tomusdrw) [inline src/api/parity_set.rs]: I don't think the bound is required

File: src/api/parity_set.rs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+    fn transport(&self) -> &T {
+        &self.transport
+    }
+}
+
+impl<T: Transport> ParitySet<T> {
+    
+    /// Set Parity to accept non-reserved peers (default behavior)
+    pub fn parity_accept_non_reserved_peers(&self) -> CallFuture<bool, T::Out> {
```
Comment (tomusdrw) [inline src/api/parity_set.rs]: Can we drop the `parity_` prefix?

File: src/api/mod.rs (hunk 1 above)
Comment (jordipainan) [reply src/api/mod.rs]: Sorry! Fixed

File: src/api/parity_set.rs (hunk 2 above)
Comment (jordipainan) [reply src/api/parity_set.rs]: Correct! Works well too if bound is not present.

File: src/api/parity_set.rs (hunk 3 above)
Comment (jordipainan) [reply src/api/parity_set.rs]: Yes! With `parity_` prefix its a bit repetitive :)

**Comment by HCastano** [comment]
Hey, just as an FYI `setMinGasPrice` isn't deprecated anymore (see https://github.com/paritytech/parity-ethereum/pull/10294).

**Comment by jordipainan** [comment]
Thank you!! ^^

**Comment by blanksteer** [comment]
Am i good to add peer listing to ``ParitySet`` or is that a function most would rather not have?

**Comment by jordipainan** [comment]
Hi! Sorry for the delay answering. In my own opinion I think `peer_listing` fits better on the `net` namespace.
I see `parity_set` as a namespace for configure some aspects of the client and `peer_listing` returns some info about the client state and it can be a bit confusing because is something regarding to the network.
What do you think ?

**Comment by CPerezz** [comment]
I agree with @jordipainan . It would probably fit better on anothre namespace like `net` where we also have the `peer_count` feature.

**Comment by tomusdrw** [comment]
I treat namespaces as structs that correspond to the first part of the RPC method name (in this case `parity_netPeers`).

Since it might make sense to logically group some methods together (like `parity_netPeers` and `parity_startNetwork`, `parity_netPort`) but that grouping would be completely arbitrary for `rust-web3`. Also it makes it really confusing where the methods should be searched for.

I'd prefer to keep a simple rule of thumb where methods are placed in the codebase, moving arbitrary methods between the namespaces doesn't seem like a good idea.

**Comment by jordipainan** [comment]
I understand your point of view, perhaps it is easier for me to find the methods the way I have described, but if in general you think that this approach can lead to confusion and does not match with what was previously developed, go ahead I trust in your criteria :)

**Pull request merged**